### ISSUE TITLE
With multiple tags for a scenario the previous tags are being overwritten

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge#410]
### Expected behavior
If there are multiple tags then it should get appended to the list

### Actual behavior
It is getting overwritten

### Steps to reproduce
1. Create a spec
```
Specification Heading
=====================
tags:top

* something1

c
-----------------------------
tags:scenario3 ppp a b c d e f g h i j k l m n o p q r s t u v w x y z

* something
tags:b
```
2. run the spec
3. The tag associated with c is only `b`

### Gauge version
```
Gauge version: 0.4.1.nightly-2016-05-18

Plugins
-------
html-report (2.1.1.nightly-2016-05-19)
java (0.4.1.nightly-2016-05-17)
```

**Comment by apoorvam** [comment]
Ideally, tags should be defined only once in a scenario. So if it's defined more than once, we should throw a parse error.

**Comment by apoorvam** [comment]
Fix should be available in nightly >= 03.06.2016
